packages/rsbuild-plugin-angular/src/lib/plugin/compilation/ast-transformations/adjust-static-class-members.ts

```diff
@@ -0,0 +1,570 @@
+
+    visitedClasses.add(classDeclaration);
+
+    const className = classDeclaration.getName();
+
+    const isDefaultExport =
+      'isDefaultExport' in classDeclaration
+        ? classDeclaration.isDefaultExport()
+        : false;
+    const isExported = classDeclaration.hasModifier(SyntaxKind.ExportKeyword);
+
+    // Collect all siblings first before any modifications
+    let analysis = exportDefaultAnalysis.get(classDeclaration);
+    if (!analysis) {
+      analysis = analyzeClassSiblings(classDeclaration, options.wrapDecorators);
+    }
+    const { shouldWrap, hasSideEffects: propHasSideEffects } =
+      analyzeClassStaticProperties(classDeclaration);
+
+    const {
+      hasPotentialSideEffects,
+      wrapNodes,
+      nodesToRemove,
+      hasOnlySideEffects,
+    } = analysis;
+
+    // Skip if only side effects
+    if (hasOnlySideEffects || hasPotentialSideEffects || propHasSideEffects) {
+      continue;
+    }
+
+    // Create the modifications array to process them in order
+    const modifications: Array<() => void> = [];
+
+    // Add removal operations first
+    nodesToRemove.forEach((node) => {
+      modifications.push(() => {
+        if (!node.wasForgotten()) {
+          node.remove();
+        }
+      });
+    });
+
+    // Handle wrapping if needed
+    if (wrapNodes.length > 0 || shouldWrap) {
+      const exportPrefix = isExported && !isDefaultExport ? 'export ' : '';
+
+      modifications.push(() => {
+        if (isDefaultExport) {
+          classDeclaration.toggleModifier('default', false);
+          classDeclaration.toggleModifier('export', false);
+          sourceFile.addExportDeclaration({
+            namedExports: [{ name: className!, alias: 'default' }],
+          });
+        } else if (isExported) {
+          classDeclaration.toggleModifier('export', false);
+        }
+      });
+
+      modifications.push(() => {
+        let wrappedClass: string;
+        let varStmt: VariableStatement | undefined;
+        if (Node.isClassExpression(classDeclaration)) {
+          // 1) Look for a VariableDeclaration ancestor
+          const variableDecl = classDeclaration.getFirstAncestorByKind(
+            SyntaxKind.VariableDeclaration
+          );
+
+          // 2) Extract the variable name (the left-hand side)
+          const variableName = variableDecl?.getName();
+
+          // 3) Call your wrap function with the variable name
+          varStmt = variableDecl
+            ?.getParentIfKind(SyntaxKind.VariableDeclarationList)
+            ?.getParentIfKind(SyntaxKind.VariableStatement);
+
+          const varDeclarationKind = varStmt?.getDeclarationKind();
+
+          wrappedClass = wrapClassWithIIFE(
+            classDeclaration,
+            wrapNodes,
+            variableName,
+            varDeclarationKind
+          );
+
+          // 4) Replace the entire variable statement text (or just the variable declaration)
+          // so the new code is `var Foo = /*#__PURE__*/ ( ... )();`
```

Comment:
It would help to give a before example. This comment later on can get transformed to a test. with the mentioned example values.